fakemails/generate-fake-mail.py

```diff
@@ -0,0 +1,216 @@
+    lmtp_id = generate_lmtp_id()
+    esmtps_id = generate_esmtps_id()
+    timestamp_date = generate_timestamp(t1, t2)
+    b_header = generate_b_header()
+    full_date = generate_full_date(int(timestamp_date))
+
+    content = template.replace("ENV_ACCOUNT_UID", account_uid)
+    content = content.replace("ENV_DOVECOT_MESSAGE_ID", dovecot_message_id)
+    content = content.replace("ENV_LMTP_ID", lmtp_id)
+    content = content.replace("ENV_ESMTPS_ID", esmtps_id)
+    content = content.replace("ENV_B_HEADER", b_header)
+    content = content.replace("ENV_FULL_DATE", full_date)
+    content = content.replace("ENV_TIMESTAMP_DATE", timestamp_date)
+    content = content.replace("ENV_ESMTPSA_ID", esmtpsa_id)
+    content = content.replace("ENV_MESSAGE_ID", message_id)
+
+    return content, dovecot_message_id, int(timestamp_date)
+
+def create_fake_mail(content, dovecot_message_id, timestamp, fake_dir):
+    filename = f"{timestamp}.M655385P9377.mailstack-dovecot-server-{dovecot_message_id},S=SIZE:2,S"
+    file_path = f"{fake_dir}/{filename}"
+
+    with open(file_path, 'w') as file:
+        file.write(content)
+
+    new_filename = filename.replace("SIZE", str(os.path.getsize(file_path)))
+    new_file_path = f"{fake_dir}/{new_filename}"
+    os.rename(file_path, new_file_path)
+    os.utime(new_file_path, (timestamp, timestamp))
+
+def convert_total_size_to_bytes(account_size):
+    match = re.match(r"(\d+)(MB|GB|TB)", account_size, re.IGNORECASE)
+    if not match:
+        raise ValueError("Invalid size format. Use the format such as '1GB'.")
+    
+    size, unit = match.groups()
+    size = float(size)
+    
+    convert_to_bytes = {
+        'MB': 1024 ** 2,
+        'GB': 1024 ** 3,
+        'TB': 1024 ** 4
+    }
+    
+    b_size = size * convert_to_bytes[unit.upper()]
+
+    return int(b_size)
+
+def get_user_infos(account):
+    try:
+        data = '[["user",{"userMask":["' + account + '"]},"tag1"]]'
+        headers = {
+            'Content-Type': 'application/json'
+        }
+
+        doveadm_url = os.getenv("API_URL", "http://mailstack-dovecot-cluster:8080")
+        doveadm_username = os.getenv("API_USERNAME", 'doveadm')
+        doveadm_password = os.getenv("API_PASSWORD")
+        url = f"{doveadm_url}/doveadm/v1"
+
+        response = requests.post(url, data=data, headers=headers, auth=HTTPBasicAuth(doveadm_username, doveadm_password), timeout=10)
+
+        if response.status_code == 200:
+            respj = json.loads(response.text)
+            uid = respj[0][1][account]['uid']
+            gid = respj[0][1][account]['gid']
+
+            return uid, gid
+        elif response.status_code == 401:
+            raise ValueError("Username or password is invalid")
+        elif response.status_code == 403:
+            raise ValueError("User does not have permission to access this resource")
+        else:
+            respj = json.loads(response.text)
```

Comment:
Using `os.path.getsize` may lead to a race condition if the file is modified between creation and size calculation. Consider capturing the size directly after writing to avoid this potential issue.
```suggestion
file.write(content)\nnew_size = file.tell()
```